### ISSUE TITLE
Throw when a value other than `cy` is returned from a test or command function

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/cypress#463]
During an `e2e` or `integration` test, we should throw a specific error message when:
  1.  a value is returned to a callback function
  2. in either a mocha `it` test or in a custom command 
  3. and a `cy` command was invoked.

We're going to opt for a stricter API up front contract instead of being magical behind the scenes and forcing return values unbeknownst to the developer.

The following situations should throw:

```js
Cypress.Commands.addParent("foo", function(){
  cy.get("body").should("have.class", "foo")

  return "foobarbaz"
})

it("should throw", function(){
  cy.get("body").should("have.class", "foo")
   
  // currently Cypress detects this and forcibly
  // returns 'cy'. Instead we should throw.
  return "foobarbaz"
})

it("should also throw", function(){
  cy.foo()
})

it("does not throw", function(){
  // this is okay since no cy commands were invoked!
  expect(true).to.be.true
})
```

**Comment by brian-mann** [comment]
Also if promises are returned that is handled by: #464

And if promises are returned in a custom command that is handled by: #435 

All other value types will cause an exception to be thrown either when returning a value to a test, or a custom command.

**Comment by brian-mann** [comment]
Error when value is returned in a test

![screen shot 2017-08-24 at 6 19 39 pm](https://user-images.githubusercontent.com/1268976/29692308-4d0184a4-88fd-11e7-872c-78d2519e257a.png)

Error when value is returned in a custom command

![screen shot 2017-08-24 at 6 51 16 pm](https://user-images.githubusercontent.com/1268976/29692312-53ff0646-88fd-11e7-9794-f58f59495821.png)

**Comment by brian-mann** [comment]
The code for this is done, but this has yet to be released. We'll update the issue and reference the changelog when it's released.

**Comment by brian-mann** [comment]
Fixed in [`0.20.0`](https://docs.cypress.io/guides/references/changelog.html#0-20-0)
